core/src/main/java/bio/terra/pearl/core/service/notification/email/SendgridClient.java

```diff
@@ -38,8 +37,8 @@ public SendgridClient(Environment env, ApplicationRoutingPaths applicationRoutin
   public String sendEmail(Mail mail) {
     if (StringUtils.isEmpty(sendGridApiKey)) {
       // if there's no API key, (likely because we're in a CI environment), don't even attempt to send an email
-      log.info("Email send skipped: no sendgrid api provided");
-      throw new UnsupportedOperationException("Attempted to send email without sendgrid key");
+      log.warn("Email send skipped: no sendgrid api provided");
+      return null;
```

Comment:
It felt weird that we were throwing a full exception on this and it made testing difficult - is a warning log and not saving the key the expected behavior here?

Reply:
good question -- it's definitely an error if this happens in prod.  So maybe have it only be an error if deploymentZone is prod, and otherwise have it be a warning